prow/cmd/crier/main.go

```diff
@@ -0,0 +1,164 @@
+func loadClusterConfig() (*rest.Config, error) {
+	clusterConfig, err := rest.InClusterConfig()
+	if err == nil {
+		return clusterConfig, nil
+	}
+
+	credentials, err := clientcmd.NewDefaultClientConfigLoadingRules().Load()
+	if err != nil {
+		return nil, fmt.Errorf("could not load credentials from config: %v", err)
+	}
+
+	clusterConfig, err = clientcmd.NewDefaultClientConfig(*credentials, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load client configuration: %v", err)
+	}
+	return clusterConfig, nil
+}
+
+// TODO(krzyzacy): move this to kube???
+// getKubernetesClient retrieves the Kubernetes cluster
+// client from within the cluster
+func getKubernetesClient() (kubernetes.Interface, prowjobclientset.Interface) {
+	config, err := loadClusterConfig()
+	if err != nil {
+		logrus.Fatalf("failed to load cluster config: %v", err)
+	}
+
+	// generate the client based off of the config
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logrus.Fatalf("getClusterConfig: %v", err)
+	}
+
+	prowjobClient, err := prowjobclientset.NewForConfig(config)
+	if err != nil {
+		logrus.Fatalf("getClusterConfig: %v", err)
+	}
+
+	logrus.Info("Successfully constructed k8s client")
+	return client, prowjobClient
+}
+
+func main() {
+	logrus.SetFormatter(
+		logrusutil.NewDefaultFieldsFormatter(nil, logrus.Fields{"component": "crier"}),
+	)
+
+	// get the Kubernetes client for connectivity
+	client, prowjobClient := getKubernetesClient()
+
+	// retrieve our custom resource informer which was generated from
+	// the code generator and pass it the custom resource client, specifying
+	// we should be looking through all namespaces for listing and watching
+	informer := prowjobinformer.NewProwJobInformer(
+		prowjobClient,
+		meta_v1.NamespaceAll,
+		0,
+		cache.Indexers{},
+	)
+
+	// create a new queue so that when the informer gets a resource that is either
+	// a result of listing or watching, we can add an idenfitying key to the queue
+	// so that it can be handled in the handler
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+	// add event handlers to handle the three types of events for resources:
+	//  - adding new resources
+	//  - updating existing resources
+	//  - deleting resources
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			// convert the resource object into a key (in this case
+			// we are just doing it in the format of 'namespace/name')
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			logrus.Infof("Add prowjob: %s", key)
+			if err == nil {
+				// add the key to the queue for the handler to get
+				queue.Add(key)
+			}
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(newObj)
+			logrus.Infof("Update prowjob: %s", key)
+			if err == nil {
+				queue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
```

Comment:
You can retrieve state from the tombstone if you need it

Reply:
crier don't care when sinker deletes prowjobs